extract/extract/extractors/hm.py

```diff
@@ -0,0 +1,219 @@
+    "Duvet recyclé": CertificateType.OTHER,
+    "Laine recyclée": CertificateType.OTHER,
+    "Laiton recyclé": CertificateType.OTHER,
+    "PCTG recyclé": CertificateType.OTHER,
+    "Papier recyclé": CertificateType.OTHER,
+    "Plumes recyclé": CertificateType.OTHER,
+    "Polystyrène recyclé": CertificateType.OTHER,
+    "Polyamide recyclé": CertificateType.OTHER,
+    "Polyester recyclé": CertificateType.OTHER,
+    "zinc recyclé": CertificateType.OTHER,
+    # other
+    "In-conversion cotton": CertificateType.OTHER,
+    "Lin biologique": CertificateType.OTHER,
+    "Lyocell issue d’une gestion forestière durable": CertificateType.OTHER,
+    "Coton biologique": CertificateType.OTHER,
+    "Coton en conversion": CertificateType.OTHER,
+}
+
+_LABEL_MAPPING_HIGG = {
+    # TODO: Add Higg levels when available in sustainability-labels.json
+    "0": CertificateType.HIGG_INDEX_MATERIALS,
+    "1": CertificateType.HIGG_INDEX_MATERIALS,
+    "2": CertificateType.HIGG_INDEX_MATERIALS,
+    "3": CertificateType.HIGG_INDEX_MATERIALS,
+}
+
+
+def _get_product_details(beautiful_soup: BeautifulSoup) -> Dict:
+    """
+    Extracts the JSON product data from script tag of HTML and returns the relevant information of
+    the scraped color variant.
+
+    The basic JSON package can not handle 'messy' JSON data, so we have to use chompjs.
+
+    Args:
+        beautiful_soup (BeautifulSoup): Parsed HTML.
+
+    Returns:
+        Dict: product data JSON
+    """
+    for script in beautiful_soup.find_all("script"):
+        if script.string:
+            if "productArticleDetails = {" in script.string:
+                data = script.string.split("var productArticleDetails = ")[1][:-2]
+                data = re.sub(r"isDesktop \? \'.*' : ", "", data)
+                data = chompjs.parse_js_object(data)
+                article_code = data.get("articleCode", "")
+                return data.get(article_code, {})
+    return {}
+
+
+def _get_higg_label(product_data: Dict) -> CertificateType:
+    """
+    Extracts the HIGG_INDEX_MATERIALS label from article product_data.
+
+    Args:
+        product_data (Dict): Product data JSON.
+
+    Returns:
+        CertificateType: CertificateType.HIGG_INDEX_MATERIALS
+    """
+    if higg_level := product_data.get("higg", {}).get("levelAchievement"):
+        return _LABEL_MAPPING_HIGG.get(higg_level)
+    return None
+
+
+def _get_hm_conscious(product_data: Dict) -> CertificateType:
+    """
+    Extracts the HM_CONSCIOUS label from article product_data.
+
+    Args:
+        product_data (Dict): Product data JSON.
+
+    Returns:
+        CertificateType: CertificateType.HM_CONSCIOUS.
+    """
+    if markers := product_data.get("marketingMarkers"):
+        for marker in markers:
+            if marker.get("text") == "Conscious choice":
+                return CertificateType.HM_CONSCIOUS
+    return None
+
+
+def _get_sustainability(beautiful_soup: BeautifulSoup, product_data: Optional[Dict]) -> List[str]:
```

Comment:
This method does not work for h&m because I am checking if a string is contained within the description of the materials. So there is no extraction of terms that can be compared against our `sustainability-labels.json`, see: https://github.com/calgo-lab/green-db/blob/991e24582ad4ff7f07bb7111b6bbd11ee18af4b3/extract/extract/extractors/hm.py#L206

Reply:
I see, added a comment to use it when higg levels are available.